src/native/corehost/fxr/hostfxr.cpp

```diff
@@ -571,6 +572,126 @@ namespace
+    fx_resolver_t::resolution_failure_info failure_info;
+    rc = app_config.get_is_framework_dependent()
+        ? fx_resolver_t::resolve_frameworks(host_info.dotnet_root, override_settings, app_config, fx_definitions, failure_info)
+        : StatusCode::Success;
+
+    if (callback)
+    {
+        std::vector<hostfxr_framework_result> resolved;
+        std::vector<hostfxr_framework_result> unresolved;
+
+        pal::string_t config_dir;
+        if (app_config.get_is_framework_dependent())
+        {
+            for (const auto& fx : fx_definitions)
+            {
+                // Skip the app itself
+                if (fx.get() == app)
+                    continue;
+
+                resolved.push_back({ sizeof(hostfxr_framework_result), fx->get_name().c_str(), fx->get_requested_version().c_str(), fx->get_found_version().c_str(), fx->get_dir().c_str() });
+            }
+
+            switch ((StatusCode)rc)
+            {
+                case StatusCode::FrameworkMissingFailure:
+                    unresolved.push_back({ sizeof(hostfxr_framework_result), failure_info.missing.get_fx_name().c_str(), failure_info.missing.get_fx_version().c_str(), nullptr, nullptr });
+                    break;
+                case StatusCode::FrameworkCompatFailure:
+                    unresolved.push_back({ sizeof(hostfxr_framework_result), failure_info.incompatible_lower.get_fx_name().c_str(), failure_info.incompatible_lower.get_fx_version().c_str(), nullptr, nullptr });
+                    unresolved.push_back({ sizeof(hostfxr_framework_result), failure_info.incompatible_higher.get_fx_name().c_str(), failure_info.incompatible_higher.get_fx_version().c_str(), nullptr, nullptr });
+                    break;
+                case StatusCode::InvalidConfigFile:
+                    assert(failure_info.invalid_config != nullptr);
+                    unresolved.push_back({ sizeof(hostfxr_framework_result), failure_info.invalid_config->get_name().c_str(), failure_info.invalid_config->get_requested_version().c_str(), failure_info.invalid_config->get_found_version().c_str(), failure_info.invalid_config->get_dir().c_str() });
+                    break;
+                default:
+                    break;
+            }
+        }
+        else
+        {
+            // For self-contained apps, add all the included frameworks as resolved frameworks assumed to be next to the config
+            config_dir = get_directory(runtime_config);
+            remove_trailing_dir_separator(&config_dir);
+            for (const fx_reference_t& fx : app_config.get_included_frameworks())
+            {
+                resolved.push_back({ sizeof(hostfxr_framework_result), fx.get_fx_name().c_str(), fx.get_fx_version().c_str(), fx.get_fx_version().c_str(), config_dir.c_str() });
+            }
+        }
+
+        const hostfxr_resolve_frameworks_result result
+        {
+            sizeof(hostfxr_resolve_frameworks_result),
```

Comment:
```suggestion
            sizeof(result),
```

The above is a bit better because it ensures if we change the type of `result` the size automatically flows.